### PR TITLE
Fb 131 authentication zero trust approach

### PR DESCRIPTION
**Pull request by jennifer-fintlabs**
Update order history, added oAuth, added github actions

### CODE REVIEW

**Review by trondsevre — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -1,12 +1,12 @@
 plugins {
-    id 'org.springframework.boot' version '2.5.5'
-    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
+    id 'org.springframework.boot' version '2.6.4'
```
Comment (fsjovatsen) [inline build.gradle:2]: Kanskje en skulle gått for Spring boot 2.7.4, latest and greatest.

File: src/main/java/no/fint/betaling/Application.java [hunk 2]
```diff
@@ -2,10 +2,12 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
 import org.springframework.cache.annotation.EnableCaching;
 import org.springframework.scheduling.annotation.EnableScheduling;
 
 @SpringBootApplication
+@ConfigurationPropertiesScan
```
Comment (fsjovatsen) [inline src/main/java/no/fint/betaling/Application.java:10]: Hva trengs den til?

File: src/main/java/no/fint/betaling/controller/FileController.java [hunk 3]
```diff
@@ -20,12 +20,15 @@
...
-    @Autowired
     private GroupService groupService;
 
+    public FileController(FileService fileService, GroupService groupService) {
+        this.fileService = fileService;
+        this.groupService = groupService;
+    }
+
     @PostMapping
     public ResponseEntity getCustomersOnFile(@RequestHeader(name = "x-school-org-id") String schoolId, @RequestBody byte[] file) throws NoVISIDColumnException, UnableToReadFileException, InsufficientDataException, HttpMediaTypeNotAcceptableException {
```
Comment (fsjovatsen) [inline src/main/java/no/fint/betaling/controller/FileController.java:33]: Burde egentlig `x-school-org-id` vært en del av api url'en?

File: src/main/java/no/fint/betaling/controller/PrincipalController.java [hunk 4]
```diff
@@ -1,8 +1,15 @@
...
+import no.vigoiks.resourceserver.security.FintJwtEndUserPrincipal;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 @CrossOrigin(origins = "*")
```
Comment (fsjovatsen) [inline src/main/java/no/fint/betaling/controller/PrincipalController.java:15]: Tror denne kan taes bort for alle controllerne.

File: src/main/java/no/fint/betaling/Application.java (hunk 2 above)
Comment (jennifer-fintlabs) [reply src/main/java/no/fint/betaling/Application.java:10]: This is necessary for the application properties (@ConfigurationProperties)  to load properly.

File: src/main/java/no/fint/betaling/controller/FileController.java (hunk 3 above)
Comment (jennifer-fintlabs) [reply src/main/java/no/fint/betaling/controller/FileController.java:33]: This is set as a TODO in the next update.

**Pull request merged**